tests/query/snql/test_snql_anonymizer.py

```diff
@@ -0,0 +1,385 @@
+          $S
+        ),
+        $N
+      ),
+      equals(
+        arrayAll(
+          b,
+          $S,
+          tuple(
+            $S,
+            $S
+          )
+        ),
+        $N
+      )
+    ),
+    $N
+  ),
+  and(
+    equals(
+      project_id,
+      $N
+    ),
+    and(
+      greaterOrEquals(
+        timestamp,
+        toDateTime(
+          $S
+        )
+      ),
+      less(
+        timestamp,
+        toDateTime(
+          $S
+        )
+      )
+    )
+  )
+)""",
+        id="Special array join functions",
+    ),
+    pytest.param(
+        f"""MATCH
+            (e: events) -[contains]-> (t: transactions),
+            (e: events) -[assigned]-> (ga: groupassignee)
+        SELECT 4-5, ga.c
+        WHERE {build_cond('e')} AND {build_cond('t')}""",
+        """MATCH LEFT LEFT e, Entity(events) TYPE JoinType.INNER RIGHT ga, Entity(groupassignee)
+ ON e.event_id ga.group_id TYPE JoinType.INNER RIGHT t, Entity(transactions)
+ ON e.event_id t.event_id SELECT minus(
+  $N,
+  $N
+), ga.c WHERE and(
+  equals(
+    e.project_id,
+    $N
+  ),
+  and(
+    greaterOrEquals(
+      e.timestamp,
+      toDateTime(
+        $S
+      )
+    ),
+    and(
+      less(
+        e.timestamp,
+        toDateTime(
+          $S
+        )
+      ),
+      and(
+        equals(
+          t.project_id,
+          $N
+        ),
+        and(
+          greaterOrEquals(
+            t.finish_ts,
+            toDateTime(
+              $S
+            )
+          ),
+          less(
+            t.finish_ts,
+            toDateTime(
+              $S
+            )
+          )
+        )
+      )
+    )
+  )
+)""",
+        id="Multi join match",
+    ),
+    pytest.param(
+        "MATCH { MATCH (events) SELECT count() AS count BY title WHERE %s } SELECT max(count) AS max_count"
+        % added_condition,
+        """MATCH (MATCH Entity(events) SELECT title, count(
+) AS `count` BY title WHERE and(
+  equals(
+    project_id,
+    $N
+  ),
+  and(
+    greaterOrEquals(
+      timestamp,
+      toDateTime(
+        $S
+      )
+    ),
+    less(
+      timestamp,
+      toDateTime(
+        $S
+      )
+    )
+  )
+)) SELECT max(
+  count
+) AS `max_count`""",
+        id="sub query match",
+    ),
+    pytest.param(
+        f"""MATCH (discover_events)
+        SELECT count() AS count BY transaction_name AS tn
+        WHERE {added_condition}
+        """,
+        """MATCH Entity(discover_events) SELECT transaction_name, count(
+) AS `count` BY transaction_name WHERE and(
+  equals(
+    project_id,
+    $N
+  ),
+  and(
+    greaterOrEquals(
+      timestamp,
+      toDateTime(
+        $S
+      )
+    ),
+    less(
+      timestamp,
+      toDateTime(
+        $S
+      )
+    )
+  )
+)""",
+        id="aliased columns in select and group by",
+    ),
+]
+
+
+@pytest.mark.parametrize("query_body, expected_snql_anonymized", test_cases)
+def test_format_expressions(query_body: str, expected_snql_anonymized: str) -> None:
```

Comment:
These tests are just a subset of tests in tests/query/snql/test_query.py. If the tests should just be in that file I can put them there, it just adds a lot of text.